linux_kernel_modules/mangoh/mangoh_red.c

```diff
@@ -179,6 +182,19 @@ static struct i2c_board_info mangoh_red_battery_charger_devinfo = {
 	I2C_BOARD_INFO("bq24190", 0x6B),
 };
 
+static struct mcp251x_platform_data mcp2515_pdata = {
```

Comment:
I would prefer to see the introduction of a "can_iot_card" module which a user can then choose to include in their SDEF if they are building a system that expects the CAN card to be plugged in.  It also looks like the corresponding remove() logic is missing which should deregister the SPI device.